app/models/event.rb

```diff
@@ -20,6 +20,62 @@ class Event < ApplicationRecord
     in_week(week).group_by { |e| e.start_time.wday }
   }
 
+  class << self
```

Comment:
both methods: not_free and occupied_slots_per_wday look like factory methods. Would be cool if you could extract them to a separate class